components/tooltip/src/vwc-tooltip-base.ts

```diff
@@ -0,0 +1,140 @@
+import {
+	html, LitElement, property, query
+} from 'lit-element';
+import { nothing, TemplateResult } from 'lit-html';
+import { classMap } from 'lit-html/directives/class-map.js';
+import type { ClassInfo } from 'lit-html/directives/class-map.js';
+import { createPopper } from '@popperjs/core';
+
+export class VWCTooltipBase extends LitElement {
+	private popperInstance: any;
+	@query('.tooltip') protected popper!: HTMLElement;
+
+	/**
+	 * @prop content - the content of the tooltip
+	 * accepts string
+	 * @public
+	 * */
+	@property({ type: String, reflect: true })
+		content?: string;
+
+	/**
+	 * @prop placement - the placement of the tooltip
+	 * accepts top, bottom, right, left
+	 * @public
+	 * */
+	@property({ type: String, reflect: true })
+		placement: | 'auto'
+		| 'auto-start'
+		| 'auto-end'
+		| 'top'
+		| 'top-start'
+		| 'top-end'
+		| 'bottom'
+		| 'bottom-start'
+		| 'bottom-end'
+		| 'right'
+		| 'right-start'
+		| 'right-end'
+		| 'left'
+		| 'left-start'
+		| 'left-end' = 'auto';
```

Comment:
isn't this ts type of values importable from popper?

Reply:
it now appears in 2 files

Reply:
@yinonov Because you want to be able to control the placement inside them both, inside the tooltip and inside the tip.